alacritty/src/display.rs

```diff
@@ -296,8 +304,12 @@ impl Display {
         update_pending: DisplayUpdate,
     ) {
         // Update font size and cell dimensions
-        if let Some(font) = update_pending.font {
-            self.update_glyph_cache(config, font);
+        if update_pending.font.is_some() || update_pending.cursor.is_some() {
+            if let Some(font) = update_pending.font {
+                self.update_glyph_cache(config, font);
+            } else if update_pending.cursor.is_some() {
+                self.clear_glyph_cache();
+            }
         }
```

Comment:
You should clear glyph_cache only once, I don't see a reason to do so twice, and for font you need `update_glyph_cache`, but for `cursor` only clear, and changing font should also update cursor anyway, since cursor is based on cell metrics, which are based on font metrics.

Reply:
ops, don't get your point initially, let me update it.